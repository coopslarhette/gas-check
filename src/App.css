.App {
  text-align: center;
  width: 100%;
}

.container {
  padding-right: 30px;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.interactionSpace {
  display: flex;
}
