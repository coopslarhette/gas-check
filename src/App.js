import React from 'react'
import './App.css'
import Container from 'react-bootstrap/Container'
import Col from 'react-bootstrap/Col'
import Row from 'react-bootstrap/Row'
import GaugeNavBar from './GaugeNavBar'
import 'bootstrap/dist/css/bootstrap.min.css'
import InputOutputArea from './InputOutputArea'
import Map from './Map'

function App() {
  return (
    <div className="App">
      <Container>
        <GaugeNavBar />
        <Row className="interactionSpace">
          <Col sm={7}>
            <InputOutputArea />
          </Col>
          <Col sm={4}>
            <Map />
          </Col>
        </Row>
      </Container>
    </div>
  )
}

export default App
