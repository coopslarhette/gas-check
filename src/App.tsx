import React, { Component } from 'react'
import './App.css'
import Container from 'react-bootstrap/Container'
import Col from 'react-bootstrap/Col'
import Row from 'react-bootstrap/Row'
import GaugeNavBar from './GaugeNavBar'
import 'bootstrap/dist/css/bootstrap.min.css'
import Map from './Map'
import InputArea from './InputArea'

type MyState = {
  origin: string; destination: string; mpg: number; gasPrice: number;
}

class App extends Component<{}, MyState> {
  constructor(props) {
    super(props)
    this.state = {
      // eslint-disable-next-line react/no-unused-state
      origin: '', destination: '', mpg: 0, gasPrice: 0,
    }
    this.handleComputeRequest = this.handleComputeRequest.bind(this)
  }

  handleComputeRequest(inputInfo: {
    origin: string; destination: string; mpg: number;
    gasPrice: number;
  }): void {
    this.setState({
      origin: inputInfo.origin,
      destination: inputInfo.destination,
      mpg: inputInfo.mpg,
      gasPrice: inputInfo.gasPrice,
    })
  }

  render(): JSX.Element {
    const { origin, destination } = this.state
    return (
      <div className="App">
        <Container>
          <GaugeNavBar />
          {/* ideally we want this to wrap the map to the bottom aat 768px width as that's the
        smallest width the input boxes still render nicely at TODO will fix later probably ;) */}
          <Row className="interactionSpace">
            <Col sm={7}>
              <InputArea handleClick={this.handleComputeRequest} />
              {/*  output area */}
            </Col>
            <Col sm={5}>
              <Map
                request={{ origin, destination, travelMode: 'DRIVING' }}
              />
            </Col>
          </Row>
        </Container>
      </div>
    )
  }
}

export default App
