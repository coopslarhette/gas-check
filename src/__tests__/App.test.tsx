import React from 'react'
import ReactDOM, { unmountComponentAtNode } from 'react-dom'
import renderer from 'react-test-renderer'
import App from '../App'

describe('App component', () => {
  let div: HTMLDivElement
  beforeEach(() => {
    div = document.createElement('div')
    document.body.appendChild(div)
  })

  afterEach(() => {
    unmountComponentAtNode(div)
    div.remove()
  })

  it('renders without crashing', () => {
    ReactDOM.render(<App />, div)
  })

  it('renders all parts correctly', () => {
    const tree = renderer
      .create(<App />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
