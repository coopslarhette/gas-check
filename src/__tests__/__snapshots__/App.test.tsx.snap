// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders all parts correctly 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <div
      className="GaugeNavBar"
    >
      <nav
        className="navbar navbar-expand navbar-light"
      >
        <a
          className="navbar-brand"
          href="https://coopslarhette.github.io/gauge/"
        >
          Gauge
        </a>
        <div
          className="ml-auto"
        >
          <a
            href="https://github.com/coopslarhette"
          >
            <img
              alt="github page"
              src="../public/github-logo.png"
            />
          </a>
        </div>
      </nav>
    </div>
    <div
      className="interactionSpace row"
    >
      <div
        className="col-sm-7"
      >
        <div
          className="flex-block"
        >
          <div
            className="row"
          >
            <div
              className="inputRow"
            >
              <div
                className="mb-3 input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                    id="basic-addon1"
                  >
                    A
                  </span>
                </div>
                <input
                  aria-describedby="basic-addon1"
                  aria-label="Input"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Enter your origin."
                />
              </div>
              <div
                className="mb-3 input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                    id="basic-addon1"
                  >
                    B
                  </span>
                </div>
                <input
                  aria-describedby="basic-addon1"
                  aria-label="Input"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Enter your destination."
                />
              </div>
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="inputRow"
            >
              <div
                className="mb-3 input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                    id="basic-addon1"
                  >
                    $
                  </span>
                </div>
                <input
                  aria-describedby="basic-addon1"
                  aria-label="Input"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Enter local cost of gas here."
                />
              </div>
              <div
                className="mb-3 input-group"
              >
                <div
                  className="input-group-prepend"
                >
                  <span
                    className="input-group-text"
                    id="basic-addon1"
                  >
                    Î·
                  </span>
                </div>
                <input
                  aria-describedby="basic-addon1"
                  aria-label="Input"
                  className="form-control"
                  onChange={[Function]}
                  placeholder="Enter you car's mpg here."
                />
              </div>
            </div>
          </div>
          <button
            className="btn btn-primary btn-lg"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Compute Cost!
          </button>
        </div>
        <div />
      </div>
      <div
        className="col-sm-5"
      >
        <div
          className="map"
        />
      </div>
    </div>
  </div>
</div>
`;
